### ISSUE TITLE
Steal features from datatest for trial

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10367]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#529|
|Type|enhancement|
|Created|2004-02-27 23:00:10Z|

<details><summary>Searchable metadata</summary>

```
trac-id__529 529
type__enhancement enhancement
reporter__itamarst itamarst
priority__low low
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1077922810000000 1077922810000000
changetime__1472326041639963 1472326041639963
version__ 
owner__ 
cc__exarkun cc__spiv cc__itamarst cc__jml cc__slyphon
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
http://formencode.org/docs/DataTest/README.html
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)|@puzzlement commented|
|-|-|

```
#!html
<pre>
Perhaps slyphon can look at this
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="slyphon's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|slyphon commented|
|-|-|

```
#!html
<pre>
interesting, i'll look a little further into this
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml commented|
|-|-|

```
#!html
<pre>
My stomping ground.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv commented|
|-|-|

```
#!html
<pre>
With the new world order where trial is actually properly pyunit compatible, if
people want to use properly designed pyunit extensions (like DataTest claims to
be), there's no need to support this directly inside trial.  It should Just Work.

jml: This can probably be marked resolved or rejected, if the above makes sense
to you.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml commented|
|-|-|

```
#!html
<pre>
It does make sense, but I'd like to hold off until the new world order comes about.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml commented|
|-|-|

```
#!html
<pre>
The original URL is broken, and Trial can now use properly designed pyunit
extensions.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_9"></a>|@rodrigc commented|
|-|-|

In changeset bf6c8661a51d7fe4fa5340a4f1f9d51e1e1d52ac
```
#!CommitTicketReference repository="" revision="bf6c8661a51d7fe4fa5340a4f1f9d51e1e1d52ac"
Merge pull request #529 from twisted/8790-rodrigc-linelog

Author: rodrigc
Reviewer: acabhishek942
Fixes: #8790

twisted.python.util.LineLog needs to be visible on Python 3
```
